src/main/java/jenkins/branch/Branch.java

```diff
@@ -42,6 +42,7 @@
 
 /**
  * A source code branch.
+ * Not to be subclassed outside this plugin.
```

Comment:
But that would cause test failures in `yaml-project` (assuming tests are run with assertions enabled). The test subclasses are harmless mocks. I think the Javadoc note suffices.